### PR TITLE
feat: implement composable method registration

### PR DESCRIPTION
**Pull request by dhargitai**
Implementation for `register` composable method.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/composables/src/types/useUser/parameters.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// TODO types
+export type TODO = any;
```
Comment (WojtekTheWebDev) [inline packages/composables/src/types/useUser/parameters.ts:2]: This type might be place in types/index.ts for now

File: packages/composables/src/useUser/actions/register.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    lastName,
+    email,
+    password,
+    acceptsMarketingEmails = true,
+    customFields = []
+  } = params;
+
+  /* eslint-disable camelcase */
+  const result = await context.$bigcommerce.api.createCustomer({
+    first_name: firstName,
```
Comment (WojtekTheWebDev) [inline packages/composables/src/useUser/actions/register.ts:25]: I'm wondering if we should map snake_case properties like this. We're passing it to the enpoints in snake_case anyway so if we do not force to use camelCase then we can import types from `@vue-storefront/bigcommerce-api `and reduce lines of code with just `const result = await context.$bigcommerce.api.createCustomer(params);`
@dhargitai what do you think about that?

Comment (dhargitai) [reply packages/composables/src/useUser/actions/register.ts:25]: @WojtekTheWebDev So you suggest to send the parameters to the composable in camel case format from the beginning by placing the mapping into the `Login.vue` file or a helper inside the theme `folder`?

I think what we'd win here would lose on the other side, but I can see the benefit of using fewer types.

Comment (WojtekTheWebDev) [reply packages/composables/src/useUser/actions/register.ts:25]: yes but I'm not sure if it should be done like this, just if we cannot get rid of camel_case in this project, we can reduce the number of types. But it's just an idea

Comment (scernisevs) [reply packages/composables/src/useUser/actions/register.ts:25]: a PR has been raised to align the lint rules to ignore camelCase rule on properties. 
Awaiting validation from Mat

Comment (dhargitai) [reply packages/composables/src/useUser/actions/register.ts:25]: I resolve this according to what we discussed on Slack.

**Review by WojtekTheWebDev — APPROVED** [review]
LGTM

**Review by scernisevs — APPROVED** [review]

**Pull request merged**